extensions/amp-fx-collection/0.1/amp-fx-collection.js

```diff
@@ -109,85 +190,87 @@ export class AmpFxCollection {
 
   /**
    * Registers an fx-enabled element with its requested fx providers.
-   * @param {!Element} fxElement
+   * @param {!Element} element
    */
-  register_(fxElement) {
-    devAssert(fxElement.hasAttribute('amp-fx'));
-    devAssert(!this.seen_.includes(fxElement));
+  register_(element) {
+    devAssert(element.hasAttribute('amp-fx'));
+    devAssert(!this.seen_.includes(element));
     devAssert(this.viewer_.isVisible());
 
-    const fxTypes = this.getFxTypes_(fxElement);
-
-    fxTypes.forEach(fxType => {
-      const fxProvider = this.getFxProvider_(fxType);
-      fxProvider.installOn(fxElement);
+    getFxTypes(element).forEach(type => {
+      this.install_(element, type);
     });
+
+    this.seen_.push(element);
   }
 
   /**
-   * Returns the array of fx types this component has specified as a
-   * space-separated list in the value of `amp-fx` attribute.
-   * e.g. `amp-fx="parallax fade-in"
-   *
-   * @param {!Element} fxElement
-   * @return {!Array<!FxType>}
+   * @param {!Element} element
+   * @param {!FxType} type
+   * @private
    */
-  getFxTypes_(fxElement) {
-    devAssert(fxElement.hasAttribute('amp-fx'));
-    const fxTypes = fxElement.getAttribute('amp-fx')
-        .trim()
-        .toLowerCase()
-        .split(/\s+/);
-
-    userAssert(fxTypes.length, 'No value provided for `amp-fx` attribute');
-
-    // Validate that we support the requested fx types.
-    fxTypes.forEach(fxType => {
-      user().assertEnumValue(FxType, fxType, 'amp-fx');
-    });
+  install_(element, type) {
+    if (scrollToggledFxTypes[type]) {
+      return installScrollToggledFx(this.ampdoc_, element, type);
```

Comment:
nit: if the `installScrollToggledFx` function doesn't return anything, the `return` here should communicate that by being on its own line